<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="x_pos" default="0.0"/>
  <arg name="y_pos" default="0.0"/>
  <arg name="z_pos" default="0.0"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find turtlebot3_gazebo)/worlds/our.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_$(arg model) -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />


  <!-- Include the turtlebot3_navigation launch file -->
  <include file="$(find turtlebot3_navigation)/launch/turtlebot3_navigation.launch">
    <arg name="map_file" value="$(env HOME)/map.yaml"/>
  </include> 

  <!-- Start the move_base node -->
  <node pkg="ass2" type="move_base.py" name="move_base_service"/>

  <!-- Start the find_objects node -->
  <node pkg="ass2" type="find_objects.py" name="find_objects"/>
  
  <!-- Start the pick_object node -->
  <node pkg="ass2" type="pick_object.py" name="pick_object"/>
  
  <!-- Start the place_object node -->
  <node pkg="ass2" type="place_object.py" name="place_object"/>

  <!-- Start the control_script node -->
  <node pkg="ass2" type="control_script.py" name="control_script"/> 

</launch>